src/library.cpp

```diff
@@ -67,11 +67,27 @@ std::shared_ptr<kiwix::Reader> Library::getReader(const QString &zimId)
     }
 }
 
-std::shared_ptr<kiwix::Searcher> Library::getSearcher(const QString &zimId)
+std::shared_ptr<zim::Archive> Library::getArchive(const QString &zimId)
 {
-    auto searcher = std::make_shared<kiwix::Searcher>();
     try {
-        searcher->add_reader(m_library.getReaderById(zimId.toStdString()).get());
+      return m_library.getArchiveById(zimId.toStdString());
+    } catch (std::out_of_range& e) {
+      return nullptr;
+    }
+}
+
+std::shared_ptr<zim::Searcher> Library::getSearcher(const QString &zimId)
+{
+    std::shared_ptr<zim::Searcher> searcher;
+    try {
+        auto archive = getArchive(zimId);
+        if (archive) {
+            if (!searcher) {
+                searcher = std::make_shared<zim::Searcher>(*archive);
+            } else {
+                searcher->addArchive(*archive);
+            }
+        }
     } catch(std::out_of_range& e) {
         return nullptr;
     }
```

Comment:
If we don't catch exception, this can be simply rewritten with :
```
auto archive = m_library.getArchiveById(zimId);
return std::make_shared<zim::Searcher>(*archive);
```
and when kiwix/libkiwix#729 will be merged (and released), we can simply use `m_library.getSearcherById(zimId)`.